### PR TITLE
Update zephyr-optee-test to be compatible with Zephyr 3.6.0

### PR DESCRIPTION
**Pull request by lorc**
There is a set of patches that enables this app to work with (at least) Zephyr 3.6.0. Tested on Spyder S4 board in baremetal setup:

```
------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [pkcs11_1000]: pass = 25, fail = 0, skip = 0, total = 25 duration = 213.773 seconds
 - PASS - [pkcs11_1000.test_1000] duration = 0.510 seconds
 - PASS - [pkcs11_1000.test_1001] duration = 0.121 seconds
 - PASS - [pkcs11_1000.test_1002] duration = 0.046 seconds
 - PASS - [pkcs11_1000.test_1003] duration = 1.706 seconds
 - PASS - [pkcs11_1000.test_1004] duration = 0.555 seconds
 - PASS - [pkcs11_1000.test_1005] duration = 0.070 seconds
 - PASS - [pkcs11_1000.test_1006] duration = 0.035 seconds
 - PASS - [pkcs11_1000.test_1007] duration = 0.051 seconds
 - PASS - [pkcs11_1000.test_1008] duration = 0.210 seconds
 - PASS - [pkcs11_1000.test_1009] duration = 0.305 seconds
 - PASS - [pkcs11_1000.test_1010] duration = 0.054 seconds
 - PASS - [pkcs11_1000.test_1011] duration = 1.104 seconds
 - PASS - [pkcs11_1000.test_1012] duration = 0.266 seconds
 - PASS - [pkcs11_1000.test_1013] duration = 1.457 seconds
 - PASS - [pkcs11_1000.test_1014] duration = 0.331 seconds
 - PASS - [pkcs11_1000.test_1015] duration = 0.744 seconds
 - PASS - [pkcs11_1000.test_1016] duration = 0.269 seconds
 - PASS - [pkcs11_1000.test_1017] duration = 0.273 seconds
 - PASS - [pkcs11_1000.test_1018] duration = 0.425 seconds
 - PASS - [pkcs11_1000.test_1019] duration = 15.369 seconds
 - PASS - [pkcs11_1000.test_1020] duration = 0.793 seconds
 - PASS - [pkcs11_1000.test_1021] duration = 37.150 seconds
 - PASS - [pkcs11_1000.test_1022] duration = 55.808 seconds
 - PASS - [pkcs11_1000.test_1023] duration = 96.107 seconds
 - PASS - [pkcs11_1000.test_1024] duration = 0.014 seconds

SUITE PASS - 100.00% [regression_1000]: pass = 20, fail = 0, skip = 0, total = 20 duration = 63.706 seconds
 - PASS - [regression_1000.test_1001] duration = 0.081 seconds
 - PASS - [regression_1000.test_1002] duration = 0.060 seconds
 - PASS - [regression_1000.test_1003] duration = 0.369 seconds
 - PASS - [regression_1000.test_1004] duration = 0.198 seconds
 - PASS - [regression_1000.test_1005] duration = 0.353 seconds
 - PASS - [regression_1000.test_1006] duration = 2.482 seconds
 - PASS - [regression_1000.test_1007] duration = 0.297 seconds
 - PASS - [regression_1000.test_1008] duration = 14.228 seconds
 - PASS - [regression_1000.test_1009] duration = 5.338 seconds
 - PASS - [regression_1000.test_1010] duration = 6.812 seconds
 - PASS - [regression_1000.test_1011] duration = 1.387 seconds
 - PASS - [regression_1000.test_1012] duration = 0.211 seconds
 - PASS - [regression_1000.test_1013] duration = 28.318 seconds
 - PASS - [regression_1000.test_1016] duration = 0.378 seconds
 - PASS - [regression_1000.test_1017] duration = 0.180 seconds
 - PASS - [regression_1000.test_1018] duration = 0.201 seconds
 - PASS - [regression_1000.test_1019] duration = 0.293 seconds
 - PASS - [regression_1000.test_1020] duration = 0.062 seconds
 - PASS - [regression_1000.test_1021] duration = 1.965 seconds
 - PASS - [regression_1000.test_1022] duration = 0.493 seconds

SUITE PASS - 100.00% [regression_4000]: pass = 20, fail = 0, skip = 0, total = 20 duration = 176.257 seconds
 - PASS - [regression_4000.test_4001] duration = 0.176 seconds
 - PASS - [regression_4000.test_4002] duration = 0.673 seconds
 - PASS - [regression_4000.test_4003] duration = 0.882 seconds
 - PASS - [regression_4000.test_4004] duration = 0.126 seconds
 - PASS - [regression_4000.test_4005] duration = 0.453 seconds
 - PASS - [regression_4000.test_4006] duration = 90.032 seconds
 - PASS - [regression_4000.test_4007_dh] duration = 1.097 seconds
 - PASS - [regression_4000.test_4007_dsa] duration = 0.216 seconds
 - PASS - [regression_4000.test_4007_ecc] duration = 3.540 seconds
 - PASS - [regression_4000.test_4007_rsa] duration = 30.919 seconds
 - PASS - [regression_4000.test_4007_symmetric] duration = 0.640 seconds
 - PASS - [regression_4000.test_4007_x25519] duration = 0.137 seconds
 - PASS - [regression_4000.test_4008] duration = 0.152 seconds
 - PASS - [regression_4000.test_4009] duration = 43.311 seconds
 - PASS - [regression_4000.test_4010] duration = 0.228 seconds
 - PASS - [regression_4000.test_4011] duration = 1.967 seconds
 - PASS - [regression_4000.test_4012] duration = 0.138 seconds
 - PASS - [regression_4000.test_4013] duration = 0.773 seconds
 - PASS - [regression_4000.test_4014] duration = 0.639 seconds
 - PASS - [regression_4000.test_4015] duration = 0.158 seconds

SUITE PASS - 100.00% [regression_4100]: pass = 14, fail = 0, skip = 0, total = 14 duration = 6.115 seconds
 - PASS - [regression_4100.test_4101] duration = 0.148 seconds
 - PASS - [regression_4100.test_4102] duration = 0.142 seconds
 - PASS - [regression_4100.test_4103] duration = 0.181 seconds
 - PASS - [regression_4100.test_4104] duration = 0.146 seconds
 - PASS - [regression_4100.test_4105] duration = 0.182 seconds
 - PASS - [regression_4100.test_4106] duration = 0.124 seconds
 - PASS - [regression_4100.test_4107] duration = 0.138 seconds
 - PASS - [regression_4100.test_4108] duration = 0.154 seconds
 - PASS - [regression_4100.test_4109] duration = 0.134 seconds
 - PASS - [regression_4100.test_4110] duration = 0.151 seconds
 - PASS - [regression_4100.test_4111] duration = 0.144 seconds
 - PASS - [regression_4100.test_4112] duration = 4.162 seconds
 - PASS - [regression_4100.test_4113] duration = 0.159 seconds
 - PASS - [regression_4100.test_4114] duration = 0.150 seconds

SUITE PASS - 100.00% [regression_6000]: pass = 19, fail = 0, skip = 0, total = 19 duration = 76.619 seconds
 - PASS - [regression_6000.test_6001] duration = 0.434 seconds
 - PASS - [regression_6000.test_6002] duration = 0.460 seconds
 - PASS - [regression_6000.test_6003] duration = 0.229 seconds
 - PASS - [regression_6000.test_6004] duration = 0.268 seconds
 - PASS - [regression_6000.test_6005] duration = 0.211 seconds
 - PASS - [regression_6000.test_6006] duration = 0.236 seconds
 - PASS - [regression_6000.test_6007] duration = 0.772 seconds
 - PASS - [regression_6000.test_6008] duration = 0.414 seconds
 - PASS - [regression_6000.test_6009] duration = 0.478 seconds
 - PASS - [regression_6000.test_6010] duration = 1.456 seconds
 - PASS - [regression_6000.test_6012] duration = 0.470 seconds
 - PASS - [regression_6000.test_6013] duration = 0.270 seconds
 - PASS - [regression_6000.test_6014] duration = 1.646 seconds
 - PASS - [regression_6000.test_6015] duration = 0.342 seconds
 - PASS - [regression_6000.test_6016] duration = 6.851 seconds
 - PASS - [regression_6000.test_6017] duration = 0.210 seconds
 - PASS - [regression_6000.test_6018] duration = 60.600 seconds
 - PASS - [regression_6000.test_6019] duration = 0.375 seconds
 - PASS - [regression_6000.test_6020] duration = 0.897 seconds

SUITE PASS - 100.00% [regression_8000]: pass = 2, fail = 0, skip = 0, total = 2 duration = 3.292 seconds
 - PASS - [regression_8000.test_8001] duration = 3.210 seconds
 - PASS - [regression_8000.test_8002] duration = 0.082 seconds

SUITE PASS - 100.00% [regression_8100]: pass = 3, fail = 0, skip = 0, total = 3 duration = 0.748 seconds
 - PASS - [regression_8100.test_8101] duration = 0.437 seconds
 - PASS - [regression_8100.test_8102] duration = 0.106 seconds
 - PASS - [regression_8100.test_8103] duration = 0.205 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION SUCCESSFUL

```

### CODE REVIEW

**Review by xakep-amatop — APPROVED** [review]
`Acked-by: Mykola Kvach <mykola_kvach@epam.com>`

**Comment by klogg** [comment]
Please carefully go through commit text. In some places while the spelling is correct words seem to be wrong. E.g. "trying to use local cope" instead of "copy" - looks sus to me

**Review by firscity — APPROVED** [review]
Thank you
`Reviewed-by: Dmytro Firsov <dmytro_firsov@epam.com>`

File: src/pkcs11_1000.c [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <utee_defines.h>
```
Comment (oleksiimoisieiev) [inline src/pkcs11_1000.c:21]: Please take into account my commit: https://github.com/xen-troops/zephyr-optee-test/pull/5/commits/7858566823b417267ed5e058f8b77c7609279bd7
Probably it's better to remove util.h

**Comment by oleksiimoisieiev** [comment]
Went through all patches. In many places changes are the same as in the PR https://github.com/xen-troops/zephyr-optee-test/pull/5. Some places are Zephyr 3.7.0 specific. Everything looks good for me.
`Reviewed-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>`

As I can see this PR was tested on spyder so I think it worth merging it. After merging I add h3ulcb support and then we can make a tag.

File: src/pkcs11_1000.c (hunk 1 above)
Comment (lorc) [reply src/pkcs11_1000.c:21]: Actually, utilities from `util.h` are used in tests. At least `ARRAY_SIZE`.

**Pull request merged**